### ISSUE TITLE
[Feature] don't stash on commit --all

### ISSUE THREAD

**Issue by root360-AndreasUlm** [pre-commit/pre-commit#1870]
Currently you have to add all files to the stage by running `git add files` before committing as pre-commit will stash all changes that are not added.
This behavior in general is fine but should not be used when the user runs `git commit --all`.
As the flag `--all` will add all changes to the commit without the need to run `git add` pre-commit should IMHO not stash in this case.

---

pre-commit version: 2.12.0
git version: 2.25.1
OS: Ubuntu 20.04

**Comment by asottile** [comment]
I can't reproduce, please show a reproduction of this issue.

**Comment by root360-AndreasUlm** [comment]
Thanks for checking, it seems I just had a configuration issue that led to my conclusion because I can't reproduce the behavior either.

I'm closing this issue for now and will reopen it, when I see this behavior again with details about it.

**Comment by root360-AndreasUlm** [comment]
I managed to reproduce the behavior I've seen but it is unrelated to pre-commit on `commit --all`.
I've written a pre-commit hook that is running `pre-commit run` within a docker container. That way I can run the same checks locally and in our CI without managing multiple configuration files.
The whole process is a bit more complex than this but the usage of `pre-commit run` results in the behaviour as it stash all unstaged changes within a patch file.

I already found the "hidden" subcommand `hook-impl` which is used by the install hook-script and am testing it to prevent the behavior.
